File: test-suite/src/test/java/io/micronaut/docs/inject/qualifiers/named/V8Engine.java
package io.micronaut.docs.inject.qualifiers.named;

import javax.inject.Singleton;

// tag::class[]
@Singleton
public class V8Engine implements Engine {
    public String start() {
        return "Starting V8";
    }

    public int getCylinders() {
        return 8;
    }

}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/inject/qualifiers/named/V8Engine.kt
package io.micronaut.docs.inject.qualifiers.named

import javax.inject.Singleton

// tag::class[]
@Singleton
class V8Engine : Engine {

    override var cylinders: Int = 8

    override fun start(): String {
        return "Starting V8"
    }

}
// end::class[]
